app/src/main/java/io/homeassistant/companion/android/widgets/media_player_controls/MediaPlayerControlsWidget.kt

```diff
@@ -392,13 +405,17 @@ class MediaPlayerControlsWidget : BaseWidgetProvider() {
         return AppDatabase.getInstance(context).mediaPlayCtrlWidgetDao().getAll().map { it.id }
     }
 
-    private suspend fun getEntity(context: Context, entityId: String, suggestedEntity: Entity<Map<String, Any>>?): Entity<Map<String, Any>>? {
+    private suspend fun getEntity(context: Context, entityIds: List<String>, suggestedEntity: Entity<Map<String, Any>>?): Entity<Map<String, Any>>? {
         val entity: Entity<Map<String, Any>>?
         try {
-            entity = if (suggestedEntity != null && suggestedEntity.entityId == entityId) {
+            entity = if (suggestedEntity != null && entityIds.contains(suggestedEntity.entityId)) {
                 suggestedEntity
             } else {
-                entityId.let { integrationUseCase.getEntity(it) }
+                val entities = entityIds
+                    .map { integrationUseCase.getEntity(it) }
+                entity = entities.firstOrNull { it?.state == "playing" }
```

Comment:
Isn't it better to do the API call inside the loop to prevent unnecessary requests for players when we've already found an active player?

Reply:
Ill rework this from a map to a loop so we only do the api calls necessary